vendor/clink.lua

```diff
@@ -397,35 +397,43 @@ local function git_prompt_filter()
         clean = get_clean_color(),
         dirty = get_dirty_color(),
         conflict = get_conflict_color(),
-        unknown = get_unknown_color()
+        nostatus = get_unknown_color()
     }
 
     local git_dir = get_git_dir()
+    local color
     cmderGitStatusOptIn = get_git_status_setting()
-
-    if git_dir then
-        -- if we're inside of git repo then try to detect current branch
-        local branch = get_git_branch(git_dir)
-        local color = colors.unknown
-        if branch then
-            if cmderGitStatusOptIn then
-                -- Has branch => therefore it is a git folder, now figure out status
-                local gitStatus = get_git_status()
-                local gitConflict = get_git_conflict()
-
-                color = colors.dirty
-                if gitStatus then
-                    color = colors.clean
-                end
-
-                if gitConflict then
-                    color = colors.conflict
-                end
-            end
-
-            clink.prompt.value = string.gsub(clink.prompt.value, "{git}", color.."("..verbatim(branch)..")")
-            return false
-        end
+    if cmderGitStatusOptIn then
+      if git_dir then
+          -- if we're inside of git repo then try to detect current branch
+          local branch = get_git_branch(git_dir)
+          if branch then
+              -- Has branch => therefore it is a git folder, now figure out status
+              local gitStatus = get_git_status()
+              local gitConflict = get_git_conflict()
+
+              color = colors.dirty
+              if gitStatus then
+                  color = colors.clean
+              end
+
+              if gitConflict then
+                  color = colors.conflict
+              end
+
+              clink.prompt.value = string.gsub(clink.prompt.value, "{git}", color.."("..verbatim(branch)..")")
+              return false
+          end
+      end
+    else
+      if git_dir then
```

Comment:
It looks possible to make a fix without duplicating code:
- Suppose the `color = colors.nostatus` line (432) is moved immediately after the `if branch then` line.
- Suppose the `if cmderGitStatusOptIn then` line is moved immediately after that (i.e. immediately after the new location of the `color = colors.nostatus` line).
- Suppose the matching `end` from the `if cmderGitStatusOptIn then` block is moved immediately before the `clink.prompt.value = ...` line.

I think that will allow the `else ...` block on line 428-436 to be removed, while still fixing the bug.

Thoughts?